impl/src/metadata.rs

```diff
@@ -92,189 +246,203 @@ impl CargoMetadataFetcher {
     Ok(crate_dir)
   }
 
-  pub fn fetch_crate_src(
+  fn inject_binaries_into_workspace(
```

Comment:
I think this needs a comment -- the name is provocative but not really self explanatory.